exchanges/bitmex/bitmex_websocket.go

```diff
@@ -66,46 +69,39 @@ const (
 	bitmexActionUpdateData  = "update"
 )
 
-var subscriptionNames = map[string]string{
-	subscription.OrderbookChannel: bitmexWSOrderbookL2,
-	subscription.AllTradesChannel: bitmexWSTrade,
+var defaultSubscriptions = subscription.List{
+	{Enabled: true, Channel: bitmexWSOrderbookL2, Asset: asset.All},
+	{Enabled: true, Channel: bitmexWSTrade, Asset: asset.All},
+	{Enabled: true, Channel: bitmexWSAffiliate, Authenticated: true},
+	{Enabled: true, Channel: bitmexWSOrder, Authenticated: true},
+	{Enabled: true, Channel: bitmexWSMargin, Authenticated: true},
+	{Enabled: true, Channel: bitmexWSTransact, Authenticated: true},
+	{Enabled: true, Channel: bitmexWSWallet, Authenticated: true},
+	{Enabled: true, Channel: bitmexWSExecution, Authenticated: true, Asset: asset.PerpetualContract},
+	{Enabled: true, Channel: bitmexWSPosition, Authenticated: true, Asset: asset.PerpetualContract},
 }
 
 // WsConnect initiates a new websocket connection
 func (b *Bitmex) WsConnect() error {
+	ctx := context.TODO()
```

Comment:
Can move this down to before `b.wsOpenStream` so it's used when it's actually needed instead of here

Reply:
Agreed.
I'm usually putting it at the top because it *should* be passed in, and declaring it is kind of representative of that.
But equally it's wasteful to declare it before error returns.
So I've flip flopped a few times between: Up at the top, After early returns, and Just In Time.
I'll move it to where it's used here.

Fixed 9f8d70449

Immediate squash incoming.